// react-js-stepper/public/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

require("./style/stepper.scss");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

var Stepper =
/*#__PURE__*/
function (_React$Component) {
  _inherits(Stepper, _React$Component);

  function Stepper(props) {
    _classCallCheck(this, Stepper);

    return _possibleConstructorReturn(this, _getPrototypeOf(Stepper).call(this, props));
  }

  _createClass(Stepper, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          steps = _this$props.steps,
          showNumber = _this$props.showNumber,
          activeStep = _this$props.activeStep,
          onSelect = _this$props.onSelect;
      var lastIndexOfSteps = steps.length - 1;
      return _react["default"].createElement(_react["default"].Fragment, null, _react["default"].createElement("div", {
        className: "stepper-container"
      }, steps.map(function (step, index) {
        return _react["default"].createElement(_react["default"].Fragment, {
          key: index
        }, _react["default"].createElement("div", {
          className: "stepper-item"
        }, _react["default"].createElement("div", {
          className: "stepper-item-outer",
          onClick: onSelect.bind(null, index + 1)
        }, _react["default"].createElement("div", {
          className: "stepper-item-inner ".concat(activeStep === index + 1 ? 'stepper-item-inner-active' : index + 1 < activeStep ? 'stepper-item-inner-completed' : 'stepper-item-inner-future')
        }, "  ", showNumber && index + 1, " ")), _react["default"].createElement("span", {
          className: "stepper-title ".concat(activeStep === index + 1 ? 'stepper-title-active' : '')
        }, " ", step.title, " ")), lastIndexOfSteps === index ? '' : _react["default"].createElement("div", {
          className: "stepper-item-outer"
        }, " "));
      })));
    }
  }]);

  return Stepper;
}(_react["default"].Component);

exports["default"] = Stepper;
Stepper.defaultProps = {
  steps: [{
    title: 'Step 1'
  }, {
    title: 'Step 2'
  }, {
    title: 'Step 3'
  }, {
    title: 'Step 4'
  }],
  showNumber: false,
  activeStep: 2,
  onSelect: function onSelect(step) {}
};
Stepper.propTypes = {
  steps: _propTypes["default"].arrayOf(_propTypes["default"].shape({
    title: _propTypes["default"].string
  })),
  showNumber: _propTypes["default"].bool,
  activeStep: _propTypes["default"].number,
  onSelect: _propTypes["default"].func
};
